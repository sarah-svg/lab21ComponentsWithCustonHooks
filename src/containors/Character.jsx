import React from 'react';

function Character() {
  return (
    <div>
           hey from a single character page 
    </div>
  );
}

export default Character;
