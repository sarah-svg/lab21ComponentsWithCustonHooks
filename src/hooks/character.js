import { createConfigItem } from '@babel/core';
import { useState, useEffect, createContext } from 'react';
import { allCharacters, characterById } from '../servicers/AirbenderApi';

export const CharacterTheme = createContext(null);


export const useCharacters = () => {
  const [loading, loaded] = useState(true);
  const [characters, setCharacters] = useState([]);

  useEffect(() => {
    allCharacters()
      .then(characters => {
        setCharacters(characters);
        loaded(false);
      });
  }, []);
  return {
    loading,
    characters
  };
};

export const useById = id => {
  const [loading, setLoading] = useState(true);
  const [character, setCharacters] = useState(null);
  useEffect(() => {
    characterById(id)
      .then(character => {
        setCharacters(character);
        setLoading(false);
      });
  }, [id]);
    
  return {
    loading,
    character
  };
};

