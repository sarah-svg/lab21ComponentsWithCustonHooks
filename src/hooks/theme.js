/* eslint-disable no-unused-vars */
import React, { useState, useEffect, createContext, useContext } from 'react';

export const CharacterTheme = createContext(null);

export const NewTheme = ({ children }) => {
  const [theme, setTheme] = useState('dark');

  return (
    <CharacterTheme.Provider value={{ theme, setTheme }}>
      {children}
    </CharacterTheme.Provider>
  );

};

export const useTheme = () => {
  const { theme, setTheme } = useContext(CharacterTheme);
  return {
    theme,
    setTheme
  };
};
