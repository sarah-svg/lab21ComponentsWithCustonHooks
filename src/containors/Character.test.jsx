import React from 'react';
import { render, screen, waitFor, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import characterApi from '../fixtures/character.json';
import Character from './Character';
import { NewTheme } from '../hooks/theme';

const server = setupServer(
  // eslint-disable-next-line max-len
  rest.get('https://last-airbender-api.herokuapp.com/api/v1/characters/', (req, res, ctx) => {
    return res(ctx.json(characterApi));
  })
);
describe('Character Page', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());
  afterEach(() => cleanup());
  it('it displays the character id page', async() => {
    render(
      <NewTheme>
        <MemoryRouter>
          <Character 
            match={{ params: { id: '5cf5679a915ecad153ab68d1' } }}/>
        </MemoryRouter>
      </NewTheme>);
    screen.getAllByText('Loading...');
    const name = await screen.findByTestId('name');

    return waitFor(() => {
      expect(name).not.toBeEmptyDOMElement();
    });
  });
  it('creates a snapshot test of the character containor', async() => {
    const { asFragment } = render(<MemoryRouter>
      <Character 
        match={{ params: { id: '5cf5679a915ecad153ab68d1' } }}/>
    </MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});

