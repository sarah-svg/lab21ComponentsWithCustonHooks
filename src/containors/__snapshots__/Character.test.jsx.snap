// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Page creates a snapshot test of the character containor 1`] = `
<DocumentFragment>
  Loading...
  <iframe
    allowfullscreen=""
    alt="Loading..."
    class="giphy-embed"
    frameborder="0"
    height="240"
    src="https://giphy.com/embed/igDPjtNl8ugFy"
    width="240"
  />
  <p>
    <a
      alt="Loading..."
      href="https://giphy.com/gifs/thecurrentsea-the-current-sea-thecurrentseala-igDPjtNl8ugFy"
    >
      via GIPHY
    </a>
  </p>
</DocumentFragment>
`;
